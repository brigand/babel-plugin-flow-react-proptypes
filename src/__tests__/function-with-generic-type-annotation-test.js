const babel = require('babel-core');
const content = `
type FooProps = {
  name: string,
}

const C = (props: FooProps) : ReactElement<*> => {
}
`;

it('function-with-generic-type-annotation', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
