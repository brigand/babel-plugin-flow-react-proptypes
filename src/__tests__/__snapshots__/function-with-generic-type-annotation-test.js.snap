// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function-with-generic-type-annotation 1`] = `
"\\"use strict\\";

var C = function C(props) {};
C.propTypes = {
  name: require(\\"prop-types\\").string.isRequired
};"
`;
